### PR TITLE
Split prescriptions-refresh CronWorkflow to multiple CronWorkflows

### PR DESCRIPTION
**Pull request by fridex**
## Related Issues and Dependencies

Fixes: https://github.com/thoth-station/thoth-application/issues/2188
Fixes: https://github.com/thoth-station/thoth-application/issues/2281

### CODE REVIEW

**Comment by Gregory-Pereira** [comment]
Haven't worked with CronWorkflows before, working on the review but might take me some time to understand the `perscription-refresh-job`.

**Comment by fridex** [comment]
Based on the discussion we had:

/assign @harshad16

**Comment by goern** [comment]
@fridex @harshad16 @codificat do we target this for this or next milestone?

**Comment by harshad16** [comment]
yes we are

File: prescriptions-refresh-job/base/argo-workflows/prescriptions-refresh-image-analysis.yaml [hunk 1]
```diff
@@ -1,110 +1,167 @@
...
+    entrypoint: image-analysis
+    templates:
+      - name: image-analysis
+        resubmitPendingPods: true
+        container:
+          name: image-analysis
+          image: prescriptions-refresh-job
+          env:
+            - name: THOTH_PRESCRIPTIONS_DRY_RUN
+              value: 1
```
Comment (harshad16) [inline prescriptions-refresh-job/base/argo-workflows/prescriptions-refresh-image-analysis.yaml:72]:
```suggestion
              value: "1"
```

int values are not accepted.

File: prescriptions-refresh-job/base/argo-workflows/prescriptions-refresh-image-analysis.yaml [hunk 2]
```diff
@@ -1,110 +1,167 @@
...
+              value: "ssh -o UserKnownHostsFile=/dev/null -o StrictHostKeyChecking=no"
+            - name: THOTH_DEPLOYMENT_NAME
+              valueFrom:
+                configMapKeyRef:
+                  name: thoth
+                  key: deployment-name
+            - name: THOTH_USER_API_HOST
+              valueFrom:
+                configMapKeyRef:
+                  name: prescription-refresh-job
```
Comment (harshad16) [inline prescriptions-refresh-job/base/argo-workflows/prescriptions-refresh-image-analysis.yaml:89]:
```suggestion
                  name: prescriptions-refresh-job
```

**Review by harshad16 — CHANGES_REQUESTED** [review]
lgtm
thanks for the awesome pr.

some minor changes are required as suggested and also we need add some more image patches.
I will try to add an extra commit in this pr to address all these changes. if it is okay with you we can proceed or feel free to pull the pr and adjust it as you feel.

File: prescriptions-refresh-job/base/argo-workflows/prescriptions-refresh-image-analysis.yaml (hunk 1 above)
Comment (harshad16) [reply prescriptions-refresh-job/base/argo-workflows/prescriptions-refresh-image-analysis.yaml]: Fixed in https://github.com/thoth-station/thoth-application/pull/2312/commits/eee94e2dc151a5a32ecf5b897ca49259e99ff63b

File: prescriptions-refresh-job/base/argo-workflows/prescriptions-refresh-image-analysis.yaml (hunk 2 above)
Comment (harshad16) [reply prescriptions-refresh-job/base/argo-workflows/prescriptions-refresh-image-analysis.yaml]: Fixed in https://github.com/thoth-station/thoth-application/pull/2312/commits/eee94e2dc151a5a32ecf5b897ca49259e99ff63b

**Comment by harshad16** [comment]
@fridex  TAL at https://github.com/thoth-station/thoth-application/pull/2312/commits/eee94e2dc151a5a32ecf5b897ca49259e99ff63b commit
I have tested it out in the test cluster it works.

we would have to monitor the mem, as the pods are running into oom.
I will do some more testing and make changes in another commit.

**Comment by fridex** [comment]
> @fridex TAL at [eee94e2](https://github.com/thoth-station/thoth-application/commit/eee94e2dc151a5a32ecf5b897ca49259e99ff63b) commit I have tested it out in the test cluster it works.
> 
> we would have to monitor the mem, as the pods are running into oom. I will do some more testing and make changes in another commit.

Perfect! 💯 Changes look really good to me, feel free to merge based on your preferences 👍🏻

**Review by harshad16 — APPROVED** [review]
thanks for the amazing pr
lgtm :100:

**Pull request merged**